client/src/components/Search/Filters/FilterList.js

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import './Filters.scss';
+import { withRouter } from 'react-router-dom';
+import { Form, Collapse } from 'react-bootstrap';
+import 'rc-slider/assets/index.css';
+import LocationSearch from './FilterTypes/LocationFilter/LocationFilter';
+import './Filters.scss';
+import search from '../../../utils/search';
+import TagsSearch from './FilterTypes/TagsFilters/TagsFilters';
+import DateFilter from './FilterTypes/DateFilter/DateFilter'
+
+class Filters extends React.Component {
+
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            newFilterType: 'location',
+            filterList: {
+                date: [ 
+                    { config:{}, params:{ 
+                        decade: 2000
+                    }}, { config:{filterMode: 'range'}, params:{
+                    start: '2000-01-25', end: '2021-01-27'}}
+                ],
+                location: [],
+                tags: [''],
+                materials: ['']
+            },
+            showFilters: true,
+        };
+    }
+
+    async handleNewFilterChange(mode) {
+         this.setState({newFilterType: mode});
+    }
+
+    async addFilter(type){
+        console.log(type)
+        this.setState(state => {
+            const daFilters = state.filterList;
+            daFilters[type].push({config:{}, params:{}});
+            return {filterList: daFilters};
+        })
+    }
+
+    async removeFilter(type, id){
+        this.setState(state => {
+            const daFilters = state.filterList;
+            daFilters[type].splice(id, 1);
+            return {filterList: daFilters};
+        })
+    }
+
+    async expand(){
+        this.setState(state => {
+            return {showFilters: !state.showFilters}
+        })
+    }
+    
+    handleDateSearchSelect(params, id) {
+        const { uri } = this.props;
+        const updatedState = this.state.filterList.date
+        updatedState[id].params = params
+        this.setState({
+            ...this.state,
+            ...updatedState
+        })
+        const try1 = {}
+        console.log(this.state.filterList)
+        for(var filterType in this.state.filterList){
+            var curPar = this.state.filterList[filterType]
+            for( var i in curPar){
+                for(let parName in curPar[i].params){
+                    if(!try1[parName]){
+                        try1[parName] = []
+                    }
+                    if(curPar[i].params[parName]) try1[parName].push(curPar[i].params[parName])
+                }
+            }
+        }
```

Comment:
I'm getting lost trying to walk myself through this.  Could you talk us through it on Thursday?